### PR TITLE
[CI:BUILD] Packit: cleanups

### PR DESCRIPTION
**Pull request by lsm5**
- remove .packit.sh
- remove all bcond_with[out] conditionals in favor of easier to read alternatives
- simplify build process by removing GOPATH and related cruft

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: rpm/podman.spec [hunk 1]
```diff
@@ -276,57 +225,33 @@ CGO_CFLAGS=$(echo $CGO_CFLAGS | sed 's/-specs=\/usr\/lib\/rpm\/redhat\/redhat-an
...
-
-mkdir _build
-cd _build
-mkdir -p src/%{provider}.%{provider_tld}/%{project}
-ln -s ../../../../ src/%{import_path}
-cd ..
-ln -s vendor src
-
 # build date. FIXME: Makefile uses '/v2/libpod', that doesn't work here?
 LDFLAGS="-X %{import_path}/libpod/define.buildInfo=$(date +%s)"
```
Comment (Luap99) [inline rpm/podman.spec:229]: I think if you build with gomodules you now have to use the `/v4/libpod` path like the makefile does. I assume without that `podman version` will not show the built date.
But please double check with the built rpm I am not sure if there is some special rpm magic that I am missing.

Comment (lsm5) [reply rpm/podman.spec]: @Luap99  updated LDFLAGS to match Makefile. In future work, I'll try to reuse the Makefile as much as possible and reduce dup. Also, what's the latest on rootlessport build with BUILDTAGS. Do we need to get it in right away ?

Comment (lsm5) [reply rpm/podman.spec]: using the LDFLAGS like what's done now did show me the built date.

Comment (Luap99) [reply rpm/podman.spec]: rootlessport should not need buildtags: https://github.com/containers/podman/pull/18875/commits/f31466e7479344d9968f6d679d9c2ef5cb45a360

**Comment by lsm5** [comment]
PTAL. Good for further review.

**Comment by lsm5** [comment]
@Luap99 @rhatdan should be good to go. PTAL.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**